File: Gombang/Weekly Contest 413/3274_Check_if_Two_Chessboard_Squares_Have_the_Same_Color.cs
// 체스판의 문자는 아스키코드 값으로 표현될 수 있기에, '아스키코드 값'과 '숫자'의 합을 통해서
// 두 수의 합이 짝수면 검은색 발판이고, 홀수면 하얀색 발판이라는 개념을 적용하여 문제를 풀이하였습니다.
public class Solution
{
	public bool CheckTwoChessboards(string coordinate1, string coordinate2)
	{
		bool firstIsBlack = (coordinate1[0] + coordinate1[1]) % 2 == 0;
		bool secondIsBlack = (coordinate2[0] + coordinate2[1]) % 2 == 0;

		return firstIsBlack == secondIsBlack;
	}
}

File: Meli/Weekly Contest 413/3274. Check if Two Chessboard Squares Have the Same Color.py
class Solution:
    def checkTwoChessboards(self, coordinate1: str, coordinate2: str) -> bool:
        col1, col2 = ord(coordinate1[0]), ord(coordinate2[0]) # convert column alphabet to number
        row1, row2 = int(coordinate1[1]), int(coordinate2[1]) 
        if col1%2 == col2%2: # 홀수 열 혹은 짝수 열끼리 같은 규칙을 가지므로 동일 규칙 열인지 판별
            if row1%2 == row2%2: # 동일 규칙 열은 행 또한 홀수 혹은 짝수 열인지 판별
                return True
            else:
                return False
        else: # 다른 규칙 열일 경우
            if row1%2 == row2%2: # 홀수/짝수 행끼리는 흑, 백 색이 다름
                return False
            return True
